File: Go/Obsidian/generator/location/location.go
package locationgenerator

import (
	"../../travel"
	"math"
	"math/rand"
)

func Generate(r *rand.Rand, centreLat float64, centreLon float64, radius float64) (float64, float64) {
	angle := r.Float64() * 2.0 * math.Pi
	rad := r.Float64() * radius
	lat := centreLat + straightline.LatitudeDistance(rad*math.Cos(angle))
	lon := centreLon + straightline.LongitudeDistance(rad*math.Sin(angle), lat)
	return lat, lon
}

File: Python/Obpysidian/generator/location.py
#!/usr/bin/python
# -*- coding: utf-8 -*-

import math
import random
import travel.straightline


class LocationGenerator:

    def generate(self, centre_latitude, centre_longitude, radius):
        angle = random.random() * 2.0 * math.pi
        radius = random.random() * radius
        latitude = centre_latitude + travel.straightline.latitude_distance(radius * math.cos(angle))
        longitude = centre_longitude + travel.straightline.longitude_distance(radius * math.sin(angle), latitude)
        return (latitude, longitude)
